domain/src/main/java/org/oppia/domain/util/StateRetriever.kt

```diff
@@ -455,39 +467,44 @@ class StateRetriever @Inject constructor(
   }
 
   // Creates a customization arg value interaction object from JSON
-  private fun createCustomizationArgValueFromJson(customizationArgValue: Any): InteractionObject {
-    val interactionObjectBuilder = InteractionObject.newBuilder()
+  private fun createCustomizationArgValueFromJson(customizationArgValue: Any): SchemaObject {
+    val schemaObjectBuilder = SchemaObject.newBuilder()
     when (customizationArgValue) {
```

Comment:
Okay, I got your point that the current code cannot differentiate between:
```
"value": {
    "content_id": "ca_placeholder_0",
    "unicode_str": "Sample unicode string"
}
```
```
"value": {
    "content_id": "ca_placeholder_0",
    "html": "Sample html string"
}
```

But the response that I have received by creating questions on localhost has unicode only in this context:

```
"customization_args": {
    "placeholder": {
        "value": {
            "content_id": "ca_placeholder_0",
            "unicode_str": ""
        }
    },
    "rows": {
        "value": 1
    }
}
```

Other than that I have seen only `content_id` and `html` and therefore I am actually not able to understand this clearly.

------------------------------------------------------------------------------------
But even if there is some case where we have something like this:
```
"choices": {
    "value": [{
        "html": "\u003cp\u003e8 + 7 + 6 + 2\u003c/p\u003e",
        "content_id": "ca_choices_0"
    }, {
        "html": "\u003cp\u003e8000 + 700 + 60 + 2\u003c/p\u003e",
        "content_id": "ca_choices_1"
    }]
}
```

Using unicode:
```
"choices": {
    "value": [{
        "unicode_str": "STR1",
        "content_id": "ca_choices_0"
    }, {
        "unicode_str": "STR2",
        "content_id": "ca_choices_1"
    }]
}
```

In above case I think the question that you are raising is how we will differentiate between the two? If so, then can we use something like this:
`if (customizationArgValue.has("unicode_str"))`
to know whether it has unicode string or not?

I am actually not 100% sure if I have understood your comment and therefore adding whatever I was able to understand.

Reply:
The unicode vs. html was just an example. My point is that there's no typing information in JSON to differentiate between which SchemaObject to choose. The only way to select a schema object type is to already know which type to select. See https://github.com/oppia/oppia/blob/develop/extensions/interactions/DragAndDropSortInput/DragAndDropSortInput.py#L37 as an example. We know from the definition of drag & drop sort input that its customization argument 'choices' will be a list type.

My suggestion is that we update StateRetriever to parse the type of schema object based on what we know interactions will define since that's the most correct way to do it, rather than trying to infer the type based on type-less JSON.

Reply:
Resolving this on behalf of Ben